ui/app/com/gu/recipeasy/AppComponents.scala

```diff
@@ -61,14 +61,13 @@ class AppComponents(context: Context)
   val googleGroupsAuthorizer = if (context.environment.mode == play.api.Mode.Prod) { new GoogleGroupsAuthorisation(configuration) } else { new GoogleGroupsAuthorisationDummy() }
 
   val healthcheckController = new Healthcheck
-  val applicationController = new Application(wsClient, configuration, db, messagesApi)
 
   val publisherController = {
     val publisherConfig = PublisherConfig(configuration, region, identity.stage)
     val contentApiClient = new ContentApi(contentApiClient = new GuardianContentClient(publisherConfig.contentAtomConfig.capiKey))
     new Publisher(wsClient, configuration, publisherConfig, db, teleporter, contentApiClient)
   }
-
+  val applicationController = new Application(wsClient, configuration, db, messagesApi, publisherController)
```

Comment:
From a design point of view, it doesn't feel right for one controller to have a dependency on another controller, but I can see why you've done it that way. I'd recommend creating a new object called something like `RecipePublisher` and moving the logic to publish from the `PublishController` inside of it. That way Both ApplicationController and PublisherController can share the use of it and you remove the need for one controller to depend on the other. Controllers are more you entry points for your routes from your API.